File: set_1/arrays/python/3c3a5515-7d4b-4950-8036-4660be6ad28b.py

def remove_dupes(arr):
    return list(set(arr))

# Solution 1:
def print_largest_three1(arr):
    arr.sort()
    arr_len = len(arr)
    return arr[arr_len-3:arr_len]


arr1 = [11, 65, 193, 36, 209, 664, 32]
res1 = print_largest_three1(arr1)
print(res1)

# Solution 2:
def print_largest_three2(arr):
    # remove the dupe elems first
    arr = remove_dupes(arr)
    arr_size = len(arr)

    if (arr_size < 3):
        return ("Invalid Input!!!")

    third = first = second = 0

    for i in range(0, arr_size):

        if (arr[i] > first):

            third = second
            second = first
            first = arr[i]

        elif (arr[i] > second):

            third = second
            second = arr[i]

        elif (arr[i] > third):
            third = arr[i]

    return ([first, second, third])


arr2 = [12, 13, 1, 10, 34, 1, 99, 99, 99, 99, 99, 99, 99, 99, 99]
res2 = print_largest_three2(arr2)
print(res2)

File: set_1/arrays/javascript/3c3a5515-7d4b-4950-8036-4660be6ad28b.js

const removeDupes = arr => {
    let tempArr = [];
    arr.forEach(element => {
        if (!tempArr.includes(element)) {
            tempArr.push(element)
        }
    });
    return tempArr;
}

// Solution 1:

const printLargestThree1 = (arr) => {
    const sortedArr = arr.sort((elem1, elem2) => {
        return elem1 - elem2
    });
    return (sortedArr.splice(sortedArr.length - 3, 3));
}
const arr1 = [10, 4, 3, 50, 23, 90];
const res1 = printLargestThree1(arr1);
console.log(res1);


// Solution 2:
// Function to print three largest elements
const printLargestThree2 = (arr) => {
    let first, second, third;
    arr = removeDupes(arr);
    arr_size = arr.length
    // There should be atleast three elements
    if (arr_size < 3) {
        return ("Invalid Input!!!");
    }

    third = first = second = -1;
    for (let i = 0; i < arr_size; i++) {
        if (arr[i] > first) {
            third = second;
            second = first;
            first = arr[i];
        }

        else if (arr[i] > second) {
            third = second;
            second = arr[i];
        }

        else if (arr[i] > third)
            third = arr[i];
    }

    return ([first, second, third])
}

// Driver code
let arr2 = [10, 4, 3, 50, 23, 90, 90, 90, 90, 90, 50, 90];
let n = arr2.length;

const res = printLargestThree2(arr2, n);
console.log(res);